wgpu-hal/src/vulkan/instance.rs

```diff
@@ -591,52 +591,52 @@ impl crate::Instance<super::Api> for super::Instance {
 
     unsafe fn create_surface(
         &self,
-        has_handle: &impl raw_window_handle::HasRawWindowHandle,
+        has_handle: &(impl raw_window_handle::HasRawWindowHandle
+              + raw_window_handle::HasRawDisplayHandle),
     ) -> Result<super::Surface, crate::InstanceError> {
+        use raw_window_handle::RawDisplayHandle;
         use raw_window_handle::RawWindowHandle;
 
-        match has_handle.raw_window_handle() {
-            RawWindowHandle::Wayland(handle)
-                if self
-                    .shared
-                    .extensions
-                    .contains(&khr::WaylandSurface::name()) =>
-            {
-                Ok(self.create_surface_from_wayland(handle.display, handle.surface))
+        match (
+            has_handle.raw_window_handle(),
+            has_handle.raw_display_handle(),
+        ) {
+            (RawWindowHandle::Wayland(handle), RawDisplayHandle::Wayland(display)) => {
+                Ok(self.create_surface_from_wayland(display.display, handle.surface))
             }
-            RawWindowHandle::Xlib(handle)
+            (RawWindowHandle::Xlib(handle), RawDisplayHandle::Xlib(display))
```

Comment:
One last comment that I missed last time - sorry.

We should give this code the same treatment as the other code with Rdh and Rwh with a sinlge-line use statement.

Reply:
Should of listened to my brain. I was thinking that shouldn't the code be matching.